src/cloudant/_common_util.py

```diff
@@ -288,3 +289,35 @@ class _Code(str):
     """
     def __new__(cls, code):
         return str.__new__(cls, code)
+
+class InfiniteSession(Session):
+    """
+    This class provides for the ability to automatically renew session login
+    information in the event of expired session authentication.
+    """
+
+    def __init__(self, username, password, server_url):
+        super(InfiniteSession, self).__init__()
+        self._username = username
+        self._password = password
+        self._server_url = server_url
+
+    def request(self, method, url, **kwargs):
+        """
+        Overrides ``requests.Session.request`` to perform a POST to the
+        _session endpoint to renew Session cookie authentication settings and
+        then retry the original request, if necessary.
+        """
+        resp = super(InfiniteSession, self).request(method, url, **kwargs)
+        path = url_parse(url).path.lower()
+        post_to_session = method.upper() == 'POST' and path == '/_session'
+        if resp.status_code == 401 and not post_to_session:
```

Comment:
I think we need to capture `403` responses here too. DBcore clusters may still return the 403 `{"error":"credentials_expired",...}` response.

I guess something along the lines of,
```python
is_expired = any((
    resp.status_code == 403 and resp.json().get('error') == 'credentials_expired',
    resp.status_code == 401
))
if not post_to_session and is_expired:
    ...
```